http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala

```diff
@@ -775,23 +774,24 @@ class AzurePubsubHandlerInterp[F[_]: Parallel](
       _ <- clusterQuery.updateClusterStatus(e.runtimeId, RuntimeStatus.Error, now).transaction
 
       auth <- samDAO.getLeoAuthToken
+      diskIdOpt <- clusterQuery.getDiskId(e.runtimeId).transaction
 
-      _ <- e.useExistingDisk match {
-        case false =>
+      _ <- (e.useExistingDisk, diskIdOpt) match {
+        // disk was supposed to be created and was
```

Comment:
So since this is within the handleAzureRuntimeCreationError function, this is only triggered when runtime creation fails for some reason and we need to clean up any extraneous resources in that case. `diskRecordOpt` will only be a `Some()` when our DB has been updated, which only happens after the disk creation message is sent to WSM --> we also need this wsmResourceId from the record to send the message to WSM.